File: mkt_fetch.js
const https = require('http');

const port = process.env.AUTH_GATEWAY_PORT || '19000';
const data = JSON.stringify({
  keyword: 'A股今日涨跌停数量 沪深300 上证指数 创业板 收盘 2026年3月24日',
  from_time: 1742745600
});

const options = {
  hostname: 'localhost',
  port: port,
  path: '/proxy/prosearch/search',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(data)
  }
};

const req = https.request(options, (res) => {
  let body = '';
  res.on('data', (chunk) => { body += chunk; });
  res.on('end', () => {
    try {
      const d = JSON.parse(body);
      const msg = d.message || JSON.stringify(d);
      // Write to a temp file using Node fs
      const fs = require('fs');
      fs.writeFileSync('C:\\Users\\admin\\.qclaw\\workspace\\mkt_raw.txt', msg, 'utf8');
      console.log('SAVED len=' + msg.length + ' success=' + d.success);
    } catch(e) {
      console.log('PARSE ERROR: ' + e.message + ' body=' + body.substring(0, 200));
    }
  });
});

req.on('error', (e) => {
  console.log('REQUEST ERROR: ' + e.message);
});

req.write(data);
req.end();

File: mkt_fetch.py
import warnings, os, json, sys, requests
warnings.filterwarnings('ignore')
os.environ['TMP'] = os.environ['TEMP'] = 'C:\\Users\\admin\\AppData\\Local\\Temp'

port = os.environ.get('AUTH_GATEWAY_PORT', '19000')
base = f'http://localhost:{port}/proxy/prosearch/search'

queries = [
    'A股今日涨停跌停数量 2026年3月24日',
    '沪深300 上证指数 创业板 今日收盘 2026年3月24日',
]

all_msgs = []
for kw in queries:
    try:
        r = requests.post(base, json={'keyword': kw, 'from_time': 1742745600}, timeout=20)
        d = r.json()
        if d.get('success') and d.get('message'):
            all_msgs.append(d['message'])
    except Exception as e:
        all_msgs.append(f'error: {e}')

combined = '\n\n---\n\n'.join(all_msgs)
# write ascii-safe
with open('C:\\Users\\admin\\.qclaw\\workspace\\mkt_raw.txt', 'w', encoding='utf-8', errors='replace') as f:
    f.write(combined)
print('done len=' + str(len(combined)))
